Semigroup-Monoid.md

```diff
@@ -0,0 +1,318 @@
+autoscale: true
+footer: ![Arrow](arrow-brand-128x128.png) [@aMateoPaolo](https://twitter.com/aMateoPaolo) [@47deg](https://twitter.com/47deg) :: [Λrrow](http://arrow-kt.io) :: [http://arrow-kt.io/docs/typeclasses/monoid/](http://arrow-kt.io/docs/typeclasses/monoid/)
+slidenumbers: true
+
+
+# Semigroup
+
+__`Semigroup`__ is available in the __arrow-typeclasses__ module.
+
+```groovy
+// gradle
+compile 'io.arrow-kt:arrow-typeclasses:$arrowVersion'
+
+import arrow.typeclasses.Semigroup
+```
+^ The Semigroup is provided on the arrow-typeclasses artifact.
+^ You can use it as soon as you add it to your build.gradle file.
+
+---
+
+# Semigroup
+
+`Semigroup` is a __Type Class__, so it defines a given behavior.
+
+`Monoid` inherit its combinators. 
+
+```
+Semigroup
+
+Monoid
+```
+
+^ Semigroup is defined as a Type Class. In pure Functional Programming, type classes define behaviors.
+^ Other very well known type classes are Monoids, and we will talk about it later in this same video since they are intimately related.
+
+---
+
+# Semigroup :: Polymorphism
+
+```kotlin
+interface Semigroup<A> {
+    //...
+}
+```
+
+- Semigroup is __parametric over a type constructor `A`__. This allows us to build abstract functions over the behaviours that Semigroup defines and forget about the concrete types that A may refer to.
```

Comment:
A is not a type constructor, just a type

Reply:
Ok, I will remove that section